packages/typeorm/src/TypeORMTransactionManager.ts

```diff
@@ -1,61 +1,109 @@
-import { Connection, EntityManager, getConnection, QueryFailedError } from 'typeorm';
-import { Operation, TransactionManager, TransactionOptions, TypeORMOptions } from '@leocode/nest-tx-core';
-import { isTypeORMTransaction, TypeORMTransaction } from './TypeORMTransaction';
-import { PostgresDriver } from 'typeorm/driver/postgres/PostgresDriver';
-import { NotATypeORMTransactionError } from './NotATypeORMTransactionError';
+import {
+  Connection,
+  EntityManager,
+  getConnection,
+  QueryFailedError,
+} from "typeorm";
+import {
+  Operation,
+  TransactionManager,
+  TransactionOptions,
+  TypeORMOptions,
+} from "@leocode/nest-tx-core";
+import { isTypeORMTransaction, TypeORMTransaction } from "./TypeORMTransaction";
+import { PostgresDriver } from "typeorm/driver/postgres/PostgresDriver";
+import { NotATypeORMTransactionError } from "./NotATypeORMTransactionError";
 
 const isRetriableError = (err: unknown, connection: Connection): boolean => {
   if (!(err instanceof QueryFailedError)) {
     return false;
   }
 
+  // NOTE: Seems that the typescript doesn't agree on changing the type of err to QueryFailedError,
+  // even if the logical operator for instanceof checkes clearly that if anything other than instanceof QueryFailedError,
+  // returns false.
+  const error = err as QueryFailedError;
```

Comment:
It worked previously, so I don't think this is the case. Which TS version do you have?

Reply:
Typescript version: 4.6.3

![image](https://user-images.githubusercontent.com/7270420/169554916-5445bd28-da91-4ff7-9751-f89170978d75.png)

Reply:
I've added a note about along:

```suggestion
  const error = err as QueryFailedError;
```

that casts the err type to the one we are looking for because the `instanceof` check doesn't (and it should do that for us).